parser/structs/AST/AST.lua

```diff
@@ -20,13 +20,24 @@ function AST:append(value)
     return child
 end
 
--- Method to print the AST for debugging purposes
-function AST:print(level)
+-- Method to print the AST in a tree-like format
+function AST:print(level, is_last)
     level = level or 0
+    is_last = is_last or true
     local prefix = string.rep("  ", level)
-    print(prefix .. "|_ " .. self.value)
-    for _, child in ipairs(self.children) do
-        child:print(level + 1)
+
+    -- Print the current node with a connecting branch
+    if level > 0 then
```

Comment:
We need the parse tree to display differently. But no worries, I will see if I can work on it tonight